RubyConf-NG/openemrs
# Add child models

#### What does this PR do?
Adds child models
#### Description of Task to be completed?
* Add a `Patient` model inheriting from `User`
* Add a `Doctor` model inheriting from `User`
* Add an `Admin` model inheriting from `User`
* Add model specs
* Add child serializers for each model
#### How should this be manually tested?
A) i) Run `rspec`
ii) * All specs should pass

B) Send:

```
curl -L -X POST 'http://localhost:3000/v1/doctors' \
-H 'Content-Type: application/json' \
--data-raw '{
    "name": "Kachi",
    "email": "onyekachi.doctor@kachi.com",
    "phone_number": "0102",
    "gender": "male",
    "date_of_birth": "1993-01-01",
    "password": "password",
    "password_confirmation": "password"
}'
```

Which should return:

```.json
{
    "data": {
        "doctor": {
            "id": "doctor-804332387bba8b7ce05fed0df2d2e2c0",
            "name": "Kachi",
            "email": "onyekachi.doctor@kachi.com",
            "phone_number": "0102",
            "date_of_birth": "1993-01-01",
            "gender": "male",
            "roles": [
                "Doctor"
            ]
        }
    },
    "extra": null
}
```

The id in the response should be prefixed with `doctor`. Should change depending on the kind of user being created

#### Any background context you want to provide?
Closes https://github.com/RubyConf-NG/openemrs/issues/8
